digdag-core/src/main/java/io/digdag/core/archive/ProjectArchive.java

```diff
@@ -63,7 +60,13 @@ public void listFiles(PathConsumer consumer)
     static void listFiles(Path projectPath, PathConsumer consumer)
         throws IOException
     {
-        listFilesRecursively(projectPath, projectPath, consumer, new HashSet<>());
+        Optional<DigdagIgnore> digdagIgnore = DigdagIgnore.ofProject(projectPath);
```

Comment:
How about moving these lines (L63-L68) to `listFilesRecursively`? I feel there is no need to add a new argument `filter` to `listFilesRecursively`. Also, it would better add comments for parsing `.digdagignore`, and files/directories that start with a dot will ignore.

Reply:
@szyn 
Thank you for your review.
> How about moving these lines (L63-L68) to `listFilesRecursively` ?

IMO, it is unnatural just a bit to parse `.digdagignore` and instanciate `DigdagIgnore` every recursive call of `listFilesRecursively`.
I would move it if you want.

Reply:
Thanks, you're right. No need to move it.